### ISSUE TITLE
Problem fetching annotations and other variables

### ISSUE THREAD

**Issue by fpopineau** [mgmeyers/obsidian-zotero-integration#193]
Hi,

I encounter the problem in #191 and #184 .
With the simple template:
```
{% for annot in annotations %}
{% if annot.annotatedText %}
> {{annot.annotatedText | nl2br}}
{% endif %}
{% endfor %}
```
I don't get any output for a PDF file that I have annotated inside Zotero (embedded PDF viewer).
The Zotero Data Explorer shows one annotation:
<img width="669" alt="image" src="https://user-images.githubusercontent.com/1915631/223492339-84644e60-4c84-45a8-9856-83c4a7ea680a.png">
and the Import-zotero preview shows the annotation:
<img width="602" alt="image" src="https://user-images.githubusercontent.com/1915631/223492553-70a043c7-6a5a-4253-b94a-bf50a40c9ffc.png">
 
I must also mention that other variables from the Zotero Data Explorer come empty, like the `pdfZoteroLink` for example.
This is with Zotero 6.0.22, Obsidian 1.1.16, Zotero Integration 2.3.6 under Windows.

Thanks for your help,

**Comment by mgmeyers** [comment]
@fpopineau Do you see any errors in obsidian's developer console when you run the import? You can open it with command/ctrl + shift + i

Also, could you run an import with this template and share the contents of the resulting markdown file?

```
Annotation count: {{annotations and annotations.length}}

---
{% for a in annotations %}
{% for k, v in a -%}
{% if k === "date" -%}
`{{k}}`: `{{v | format()}}`
{% else -%}
`{{k}}`: `{{v}}`
{% endif -%}
{% endfor %}
---
{% endfor %}
```

**Comment by fpopineau** [comment]
Thanks for the prompt answer.
No errors in the console.

The result for your template is empty:
<img width="648" alt="image" src="https://user-images.githubusercontent.com/1915631/223548184-d78b9836-5519-428a-b486-c96f965f4e56.png">

It is the same for some variables reported by the Zotero Data Explorer: variables are displayed there, but empty when I try to use them in a template.

**Comment by mgmeyers** [comment]
Hmm, strange.... Could you share a screenshot of how your import format is set up eg:

<img width="730" alt="Screenshot 2023-03-07 at 12 59 44 PM" src="https://user-images.githubusercontent.com/2694747/223551300-da4a05c1-2f29-4ab1-9266-abc00a84fd69.png">

**Comment by fpopineau** [comment]
Ok, so I am afraid I have misunderstood something here.

This does work if I use the template in an **import format**: I get the annotations.
I do not get the annotations, but only a subset of what Zotero Data Explorer reports, if I use the template in a **citation format**.
Does that make sense? Is it expected?

**Comment by mgmeyers** [comment]
Ah, I see! This is indeed intended. Citation formats do not extract annotations from PDFs, only the import formats do. If you want access to annotations, you'll need to create an import format.

**Comment by fpopineau** [comment]
Ok! Thanks a lot.
I may have been mislead by the fact that before playing with citations, I tried to make a citation format with a link to the pdf attachment (if any). And that is when I noticed that variables like `pdfZoteroLink` came empty.
Is it expected also?
I started to suspect either a bug or some bad interaction with something else (especially after seeing #184 and #191)
I can understand that annotations are not extracted for citations because that require to run your external tool (IIUC)
